python/ray/data/_internal/execution/autoscaling_requester.py

```diff
@@ -0,0 +1,91 @@
+import time
+from typing import Dict
+
+import ray
+from ray.data.context import DatasetContext
+from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
+
+
+@ray.remote(num_cpus=0)
```

Comment:
also max_retries=-1?

Reply:
You mean max_task_retries since this is an actor? To use that it seems we also need max_restarts, but we don' checkpoint the `_resource_requests` so restarting this actor cannot recover it.

Reply:
Yes, we need to make this actor restartable. We don't need to checkpoint the resources; it's ok to restart from empty dict.